afkak/consumer.py

```diff
@@ -180,7 +185,10 @@ def __init__(self, client, topic, partition, processor,
             raise ValueError(
                 'request_retry_max_attempts must be non-negative integer')
         self._fetch_attempt_count = 1
-
+        if not isinstance(auto_offset_reset, str) or auto_offset_reset not in ['earliest', 'latest', 'fail']:
+            raise ValueError(
+                'auto_offset_reset must be in \'earliest\', \'latest\', \'fail\'')
```

Comment:
Switching quotes is usually cleaner:
`"auto_offset_reset must be in 'earliest', 'latest', 'fail'"`